From 50ebae5408f22d3dfcdddeb0fc122a422c557a8e Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Sun, 24 Apr 2022 20:37:21 -0400
Subject: Backport MinGW Resource Compiler changes from 6.0

And, modifiy the changes to work with clang64. Also, remove Win10
manifest ID because it cause failure to detect openGL. This results in
"Could not use OpenGL, falling back to software rendering" being
displayed.

---
 CMakeLists.txt                                | 55 +++++++++++++++++++
 CMakeModules/MinGWResourceCompiler.cmake      | 14 ++++-
 .../msw}/bitmap2component.rc                  |  2 +-
 {cvpcb => resources/msw}/cvpcb.rc             |  2 +-
 {eeschema => resources/msw}/eeschema.rc       |  2 +-
 {gerbview => resources/msw}/gerbview.rc       |  2 +-
 {kicad => resources/msw}/kicad.rc             |  2 +-
 resources/msw/kiwin32.rc                      | 27 +++++++++
 resources/msw/manifests/x64.manifest          | 41 ++++++++++++++
 .../msw/manifests/x64_dpi_aware_pmv2.manifest | 37 +++++++++++++
 resources/msw/manifests/x86.manifest          | 41 ++++++++++++++
 .../msw/manifests/x86_dpi_aware_pmv2.manifest | 37 +++++++++++++
 .../msw}/pcb_calculator.rc                    |  2 +-
 {pcbnew => resources/msw}/pcbnew.rc           |  2 +-
 .../msw}/pl_editor.rc                         |  2 +-
 15 files changed, 258 insertions(+), 10 deletions(-)
 rename {bitmap2component => resources/msw}/bitmap2component.rc (76%)
 rename {cvpcb => resources/msw}/cvpcb.rc (69%)
 rename {eeschema => resources/msw}/eeschema.rc (71%)
 rename {gerbview => resources/msw}/gerbview.rc (71%)
 rename {kicad => resources/msw}/kicad.rc (69%)
 create mode 100644 resources/msw/kiwin32.rc
 create mode 100644 resources/msw/manifests/x64.manifest
 create mode 100644 resources/msw/manifests/x64_dpi_aware_pmv2.manifest
 create mode 100644 resources/msw/manifests/x86.manifest
 create mode 100644 resources/msw/manifests/x86_dpi_aware_pmv2.manifest
 rename {pcb_calculator => resources/msw}/pcb_calculator.rc (74%)
 rename {pcbnew => resources/msw}/pcbnew.rc (70%)
 rename {pagelayout_editor => resources/msw}/pl_editor.rc (76%)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a009957695..3dc54389df 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,6 +107,10 @@ option( KICAD_SPICE
     "Build KiCad with internal Spice simulator."
     ON )
 
+option( KICAD_WIN32_DPI_AWARE
+    "Turn on DPI awareness for Windows builds only"
+    OFF )
+
 # when option KICAD_SCRIPTING OR KICAD_SCRIPTING_MODULES is enabled:
 # PYTHON_EXECUTABLE can be defined when invoking cmake
 # ( use -DPYTHON_EXECUTABLE=<python path>/python.exe or python2 )
@@ -231,6 +235,51 @@ if( WIN32 )
     # define UNICODE and_UNICODE definition on Windows. KiCad uses unicode.
     # Both definitions are required
     add_definitions(-DUNICODE -D_UNICODE)
+
+    # In fully standards-compliant mode, M_PI et al. are defined only if
+    # _USE_MATH_DEFINES is set.
+    add_definitions( -D_USE_MATH_DEFINES )
+
+    # In fully standards-compliant mode, M_PI et al. are defined only if
+    # _USE_MATH_DEFINES is set.
+    add_definitions( -D_USE_MATH_DEFINES )
+
+    # Used for the resource compiler and other arch dependent steps
+    if( MSVC )
+        # CMake does not set CMAKE_SYSTEM_PROCESSOR correctly for MSVC
+        # and it will always return the host instead of the target arch
+        if("${MSVC_C_ARCHITECTURE_ID}" STREQUAL "ARM64")
+            set( KICAD_BUILD_ARCH "arm64" )
+            set( KICAD_BUILD_ARCH_ARM64 1 )
+        elseif("${MSVC_C_ARCHITECTURE_ID}" STREQUAL "ARM")
+            set( KICAD_BUILD_ARCH "arm" )
+            set( KICAD_BUILD_ARCH_ARM 1 )
+        elseif("${MSVC_C_ARCHITECTURE_ID}" STREQUAL "X86")
+            set( KICAD_BUILD_ARCH "x86" )
+            set( KICAD_BUILD_ARCH_X86 1 )
+            add_definitions( -DKICAD_BUILD_ARCH_X86 )
+        elseif("${MSVC_C_ARCHITECTURE_ID}" STREQUAL "x64")
+            set( KICAD_BUILD_ARCH "x64" )
+            set( KICAD_BUILD_ARCH_X64 1 )
+        else()
+            message(FATAL_ERROR "Failed to determine the MSVC target architecture: ${MSVC_C_ARCHITECTURE_ID}")
+        endif()
+    else()
+        if ( NOT CMAKE_SIZEOF_VOID_P EQUAL 8 )
+            set( KICAD_BUILD_ARCH "x86" )
+            set( KICAD_BUILD_ARCH_X86 1 )
+            add_definitions( -DKICAD_BUILD_ARCH_X86 )
+        elseif( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+            set( KICAD_BUILD_ARCH "x64" )
+            set( KICAD_BUILD_ARCH_X64 1 )
+        endif()
+    endif()
+
+    add_definitions( -DKICAD_BUILD_ARCH=${KICAD_BUILD_ARCH} )
+
+    if( KICAD_WIN32_DPI_AWARE )
+        add_definitions( -DKICAD_WIN32_DPI_AWARE=1 )
+    endif()
 endif()
 
 
@@ -301,6 +348,14 @@ if( CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang" )
     endif()
 
     if( MINGW )
+        list(APPEND mingw_resource_compiler_INCLUDE_DIRS ${CMAKE_SOURCE_DIR}/resources/msw/ )
+        list(APPEND mingw_resource_compiler_DEFINES KICAD_BUILD_ARCH=${KICAD_BUILD_ARCH} )
+        if( KICAD_BUILD_ARCH_X86 )
+            list(APPEND mingw_resource_compiler_DEFINES KICAD_BUILD_ARCH_X86=1 )
+        endif()
+        if( KICAD_WIN32_DPI_AWARE )
+            list(APPEND mingw_resource_compiler_DEFINES KICAD_WIN32_DPI_AWARE=1 )
+        endif()
         set( CMAKE_EXE_LINKER_FLAGS_RELEASE "-s" )
 
         # Since version 2.8.5, Cmake uses a response file (.rsp) to
diff --git a/CMakeModules/MinGWResourceCompiler.cmake b/CMakeModules/MinGWResourceCompiler.cmake
index 0e13ee5539..5007f94409 100644
--- a/CMakeModules/MinGWResourceCompiler.cmake
+++ b/CMakeModules/MinGWResourceCompiler.cmake
@@ -12,7 +12,7 @@ macro(mingw_resource_compiler _NAME)
     dbg_msg("CMAKE_RC_COMPILER: ${CMAKE_RC_COMPILER}")
 
     # Input file.
-    set(_IN "${CMAKE_CURRENT_SOURCE_DIR}/${_NAME}.rc")
+    set(_IN "${CMAKE_SOURCE_DIR}/resources/msw/${_NAME}.rc")
     dbg_msg("_IN: ${_IN}")
 
     # Output file.
@@ -24,10 +24,20 @@ macro(mingw_resource_compiler _NAME)
     foreach(wx_include_dir ${wxWidgets_INCLUDE_DIRS})
         set(_WINDRES_INCLUDE_DIRS ${_WINDRES_INCLUDE_DIRS} -I${wx_include_dir})
     endforeach(wx_include_dir ${wxWidgets_INCLUDE_DIRS})
+
+    foreach(_mingw_rc_include_dir ${mingw_resource_compiler_INCLUDE_DIRS})
+        set(_WINDRES_INCLUDE_DIRS ${_WINDRES_INCLUDE_DIRS} -I${_mingw_rc_include_dir})
+    endforeach()
     dbg_msg("_WINDRES_INCLUDE_DIRS: ${_WINDRES_INCLUDE_DIRS}")
 
+
+    foreach(_mingw_rc_define ${mingw_resource_compiler_DEFINES})
+        set(_WINDRES_DEFINES ${_WINDRES_DEFINES} -D${_mingw_rc_define})
+    endforeach()
+    dbg_msg("_WINDRES_DEFINES: ${_WINDRES_DEFINES}")
+
     # windres arguments.
-    set(_ARGS ${_WINDRES_INCLUDE_DIRS} -i${_IN} -o${_OUT})
+    set(_ARGS ${_WINDRES_INCLUDE_DIRS} ${_WINDRES_DEFINES} -i${_IN} -o${_OUT})
     dbg_msg("_ARGS: ${_ARGS}")
 
     # Compile resource file.
diff --git a/bitmap2component/bitmap2component.rc b/resources/msw/bitmap2component.rc
similarity index 76%
rename from bitmap2component/bitmap2component.rc
rename to resources/msw/bitmap2component.rc
index 37c3b72bef..ed8a87598d 100644
--- a/bitmap2component/bitmap2component.rc
+++ b/resources/msw/bitmap2component.rc
@@ -1,3 +1,3 @@
 icon_bitmap2component ICON "../bitmaps_png/icons/icon_bitmap2component.ico"
 
-#include "wx/msw/wx.rc"
+#include "kiwin32.rc"
diff --git a/cvpcb/cvpcb.rc b/resources/msw/cvpcb.rc
similarity index 69%
rename from cvpcb/cvpcb.rc
rename to resources/msw/cvpcb.rc
index 4c06802a27..e903e0cf40 100644
--- a/cvpcb/cvpcb.rc
+++ b/resources/msw/cvpcb.rc
@@ -1,2 +1,2 @@
 icon_cvpcb ICON "../bitmaps_png/icons/icon_cvpcb.ico"
-#include "wx/msw/wx.rc"
+#include "kiwin32.rc"
diff --git a/eeschema/eeschema.rc b/resources/msw/eeschema.rc
similarity index 71%
rename from eeschema/eeschema.rc
rename to resources/msw/eeschema.rc
index 6c12ddd863..6a3cc69a86 100644
--- a/eeschema/eeschema.rc
+++ b/resources/msw/eeschema.rc
@@ -1,3 +1,3 @@
 icon_eeschema ICON "../bitmaps_png/icons/icon_eeschema.ico"
 
-#include "wx/msw/wx.rc"
+#include "kiwin32.rc"
diff --git a/gerbview/gerbview.rc b/resources/msw/gerbview.rc
similarity index 71%
rename from gerbview/gerbview.rc
rename to resources/msw/gerbview.rc
index 05e745b8f0..352e724a0d 100644
--- a/gerbview/gerbview.rc
+++ b/resources/msw/gerbview.rc
@@ -1,3 +1,3 @@
 icon_gerbview ICON "../bitmaps_png/icons/icon_gerbview.ico"
 
-#include "wx/msw/wx.rc"
+#include "kiwin32.rc"
diff --git a/kicad/kicad.rc b/resources/msw/kicad.rc
similarity index 69%
rename from kicad/kicad.rc
rename to resources/msw/kicad.rc
index 578d82c465..ce1a18a8ba 100644
--- a/kicad/kicad.rc
+++ b/resources/msw/kicad.rc
@@ -1,3 +1,3 @@
 icon_kicad ICON "../bitmaps_png/icons/icon_kicad.ico"
 
-#include "wx/msw/wx.rc"
+#include "kiwin32.rc"
diff --git a/resources/msw/kiwin32.rc b/resources/msw/kiwin32.rc
new file mode 100644
index 0000000000..32f8766d18
--- /dev/null
+++ b/resources/msw/kiwin32.rc
@@ -0,0 +1,27 @@
+// Turn off wxWidgets attempting to use its own manifests
+#define wxUSE_NO_MANIFEST 1
+#include "wx/msw/wx.rc"
+
+#ifdef ISOLATION_AWARE_ENABLED
+#define MANIFEST_ID ISOLATIONAWARE_MANIFEST_RESOURCE_ID
+#else
+#define MANIFEST_ID CREATEPROCESS_MANIFEST_RESOURCE_ID
+#endif
+
+#if !defined( KICAD_WIN32_DPI_AWARE ) || KICAD_WIN32_DPI_AWARE == 0
+#ifdef KICAD_BUILD_ARCH_X86
+#define MANIFEST_FILE "manifests/x86.manifest"
+#else // KICAD_BUILD_ARCH_X86
+#define MANIFEST_FILE "manifests/x64.manifest"
+#endif // KICAD_BUILD_ARCH_X86
+#else // !defined( KICAD_WIN32_DPI_AWARE ) || KICAD_WIN32_DPI_AWARE == 0
+#ifdef KICAD_BUILD_ARCH_X86
+#define MANIFEST_FILE "manifests/x64_dpi_aware_pmv2.manifest
+#else // KICAD_BUILD_ARCH_X86
+#define MANIFEST_FILE "manifests/x64_dpi_aware_pmv2.manifest
+#endif // KICAD_BUILD_ARCH_X86
+#endif // !defined( KICAD_WIN32_DPI_AWARE ) || KICAD_WIN32_DPI_AWARE == 0
+
+MANIFEST_ID RT_MANIFEST MANIFEST_FILE
+
+//RESOURCE FILES MUST CONTAIN A BLANK LINE AT THE END
diff --git a/resources/msw/manifests/x64.manifest b/resources/msw/manifests/x64.manifest
new file mode 100644
index 0000000000..a5f27f64b9
--- /dev/null
+++ b/resources/msw/manifests/x64.manifest
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0" xmlns:asmv3="urn:schemas-microsoft-com:asm.v3">
+  <assemblyIdentity
+      version="0.64.1.0"
+      processorArchitecture="amd64"
+      name="Controls"
+      type="win32"
+  />
+  <dependency>
+      <dependentAssembly>
+          <assemblyIdentity
+              type="win32"
+              name="Microsoft.Windows.Common-Controls"
+              version="6.0.0.0"
+              processorArchitecture="amd64"
+              publicKeyToken="6595b64144ccf1df"
+              language="*"
+          />
+      </dependentAssembly>
+  </dependency>
+  <asmv3:application>
+    <asmv3:windowsSettings>
+      <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+      <longPathAware xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">true</longPathAware>
+    </asmv3:windowsSettings>
+  </asmv3:application>
+  <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
+    <application>
+      <!-- Windows Vista -->
+      <supportedOS Id="{e2011457-1546-43c5-a5fe-008deee3d3f0}" />
+      <!-- Windows 7 -->
+      <supportedOS Id="{35138b9a-5d96-4fbd-8e2d-a2440225f93a}" />
+      <!-- Windows 8 -->
+      <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}" />
+      <!-- Windows 8.1 -->
+      <supportedOS Id="{1f676c76-80e1-4239-95bb-83d0f6d0da78}" />
+      <!-- Windows 10 ID causes openGL to not be detected on -->
+      <!-- some Win10 machines using Intel GPUs -->
+    </application>
+  </compatibility>
+</assembly>
\ No newline at end of file
diff --git a/resources/msw/manifests/x64_dpi_aware_pmv2.manifest b/resources/msw/manifests/x64_dpi_aware_pmv2.manifest
new file mode 100644
index 0000000000..34e0ffc8e9
--- /dev/null
+++ b/resources/msw/manifests/x64_dpi_aware_pmv2.manifest
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0" xmlns:asmv3="urn:schemas-microsoft-com:asm.v3">
+  <dependency>
+    <dependentAssembly>
+      <assemblyIdentity type="win32" name="Microsoft.Windows.Common-Controls" version="6.0.0.0" processorArchitecture="amd64" publicKeyToken="6595b64144ccf1df" language="*" />
+    </dependentAssembly>
+  </dependency>
+  <trustInfo xmlns="urn:schemas-microsoft-com:asm.v3">
+    <security>
+      <requestedPrivileges>
+        <requestedExecutionLevel level="asInvoker" uiAccess="false" />
+      </requestedPrivileges>
+    </security>
+  </trustInfo>
+  <asmv3:application>
+    <asmv3:windowsSettings>
+      <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+      <dpiAwareness xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">PerMonitorV2, system</dpiAwareness>
+      <dpiAware xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">true</dpiAware>
+      <longPathAware xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">true</longPathAware>
+	</asmv3:windowsSettings>
+  </asmv3:application>
+  <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
+    <application>
+      <!-- Windows Vista -->
+      <supportedOS Id="{e2011457-1546-43c5-a5fe-008deee3d3f0}" />
+      <!-- Windows 7 -->
+      <supportedOS Id="{35138b9a-5d96-4fbd-8e2d-a2440225f93a}" />
+      <!-- Windows 8 -->
+      <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}" />
+      <!-- Windows 8.1 -->
+      <supportedOS Id="{1f676c76-80e1-4239-95bb-83d0f6d0da78}" />
+      <!-- Windows 10 ID causes openGL to not be detected on -->
+      <!-- some Win10 machines using Intel GPUs -->
+    </application>
+  </compatibility>
+</assembly>
\ No newline at end of file
diff --git a/resources/msw/manifests/x86.manifest b/resources/msw/manifests/x86.manifest
new file mode 100644
index 0000000000..2c8b801f87
--- /dev/null
+++ b/resources/msw/manifests/x86.manifest
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0" xmlns:asmv3="urn:schemas-microsoft-com:asm.v3">
+  <assemblyIdentity
+      version="0.64.1.0"
+      processorArchitecture="x86"
+      name="Controls"
+      type="win32"
+  />
+  <dependency>
+      <dependentAssembly>
+          <assemblyIdentity
+              type="win32"
+              name="Microsoft.Windows.Common-Controls"
+              version="6.0.0.0"
+              processorArchitecture="X86"
+              publicKeyToken="6595b64144ccf1df"
+              language="*"
+          />
+      </dependentAssembly>
+  </dependency>
+  <asmv3:application>
+    <asmv3:windowsSettings>
+      <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+      <longPathAware xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">true</longPathAware>
+    </asmv3:windowsSettings>
+  </asmv3:application>
+  <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
+    <application>
+      <!-- Windows Vista -->
+      <supportedOS Id="{e2011457-1546-43c5-a5fe-008deee3d3f0}" />
+      <!-- Windows 7 -->
+      <supportedOS Id="{35138b9a-5d96-4fbd-8e2d-a2440225f93a}" />
+      <!-- Windows 8 -->
+      <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}" />
+      <!-- Windows 8.1 -->
+      <supportedOS Id="{1f676c76-80e1-4239-95bb-83d0f6d0da78}" />
+      <!-- Windows 10 ID causes openGL to not be detected on -->
+      <!-- some Win10 machines using Intel GPUs -->
+    </application>
+  </compatibility>
+</assembly>
\ No newline at end of file
diff --git a/resources/msw/manifests/x86_dpi_aware_pmv2.manifest b/resources/msw/manifests/x86_dpi_aware_pmv2.manifest
new file mode 100644
index 0000000000..8c2d961862
--- /dev/null
+++ b/resources/msw/manifests/x86_dpi_aware_pmv2.manifest
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0" xmlns:asmv3="urn:schemas-microsoft-com:asm.v3">
+  <dependency>
+    <dependentAssembly>
+      <assemblyIdentity type="win32" name="Microsoft.Windows.Common-Controls" version="6.0.0.0" processorArchitecture="x86" publicKeyToken="6595b64144ccf1df" language="*" />
+    </dependentAssembly>
+  </dependency>
+  <trustInfo xmlns="urn:schemas-microsoft-com:asm.v3">
+    <security>
+      <requestedPrivileges>
+        <requestedExecutionLevel level="asInvoker" uiAccess="false" />
+      </requestedPrivileges>
+    </security>
+  </trustInfo>
+  <asmv3:application>
+    <asmv3:windowsSettings>
+      <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
+      <dpiAwareness xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">PerMonitorV2, system</dpiAwareness>
+      <dpiAware xmlns="http://schemas.microsoft.com/SMI/2005/WindowsSettings">true</dpiAware>
+      <longPathAware xmlns="http://schemas.microsoft.com/SMI/2016/WindowsSettings">true</longPathAware>
+    </asmv3:windowsSettings>
+  </asmv3:application>
+  <compatibility xmlns="urn:schemas-microsoft-com:compatibility.v1">
+    <application>
+      <!-- Windows Vista -->
+      <supportedOS Id="{e2011457-1546-43c5-a5fe-008deee3d3f0}" />
+      <!-- Windows 7 -->
+      <supportedOS Id="{35138b9a-5d96-4fbd-8e2d-a2440225f93a}" />
+      <!-- Windows 8 -->
+      <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}" />
+      <!-- Windows 8.1 -->
+      <supportedOS Id="{1f676c76-80e1-4239-95bb-83d0f6d0da78}" />
+      <!-- Windows 10 ID causes openGL to not be detected on -->
+      <!-- some Win10 machines using Intel GPUs -->
+    </application>
+  </compatibility>
+</assembly>
\ No newline at end of file
diff --git a/pcb_calculator/pcb_calculator.rc b/resources/msw/pcb_calculator.rc
similarity index 74%
rename from pcb_calculator/pcb_calculator.rc
rename to resources/msw/pcb_calculator.rc
index 2695e79589..0bfdb74df0 100644
--- a/pcb_calculator/pcb_calculator.rc
+++ b/resources/msw/pcb_calculator.rc
@@ -1,3 +1,3 @@
 icon_pcbcalculator ICON "../bitmaps_png/icons/icon_pcbcalculator.ico"
 
-#include "wx/msw/wx.rc"
+#include "kiwin32.rc"
diff --git a/pcbnew/pcbnew.rc b/resources/msw/pcbnew.rc
similarity index 70%
rename from pcbnew/pcbnew.rc
rename to resources/msw/pcbnew.rc
index f568bb0391..5d7c800161 100644
--- a/pcbnew/pcbnew.rc
+++ b/resources/msw/pcbnew.rc
@@ -1,3 +1,3 @@
 icon_pcbnew ICON "../bitmaps_png/icons/icon_pcbnew.ico"
 
-#include "wx/msw/wx.rc"
+#include "kiwin32.rc"
diff --git a/pagelayout_editor/pl_editor.rc b/resources/msw/pl_editor.rc
similarity index 76%
rename from pagelayout_editor/pl_editor.rc
rename to resources/msw/pl_editor.rc
index 885f0e6b53..ef394934f7 100644
--- a/pagelayout_editor/pl_editor.rc
+++ b/resources/msw/pl_editor.rc
@@ -1,3 +1,3 @@
 icon_pagelayout_editor ICON "../bitmaps_png/icons/icon_pagelayout_editor.ico"
 
-#include "wx/msw/wx.rc"
+#include "kiwin32.rc"
-- 
